pkg/kubelet/cm/memorymanager/fake_memory_manager.go

```diff
@@ -28,45 +30,51 @@ import (
 )
 
 type fakeManager struct {
-	state state.State
+	// Logger is for functions where context is not passed
+	logger klog.Logger
```

Comment:
same open question, not really blocking but I haven't made my mind yet. When is it OK to store a logger instance and where is not? I need to re-read the contextual logging docs.

Reply:
I couldn't find a better way to preserve the logs when context is not plumbed all the way up. Thought storing the logger temporarily should be okay. Once we have context passed to add the methods we can remove this logger.  

Completely open to feedback in this area! Will also in the docs and see if there are recommendations for handling scenarios like this.

Reply:
I reviewed the documentation. I can infer by this text: `key idea is that libraries are passed a logger instance by their caller and use that for logging instead of accessing a global logger`  that the recommendation is to pass loggers into functions or derive them using `klog.FromContext(ctx)`. However,  none of the resources explicitly address temporary storage of loggers. Since the context is not yet plumbed through the entire call stack, temporarily storing a logger can help preserve logs until the code is fully refactored. Would it help if we document with a  comment that the logger storage here is temporary and will be removed once the context is plumbed in?

Reply:
Also,  after going through the code base, I saw a lot of places where logger instance is being stored like we are doing.

Reply:
> But the "Logger is for functions where context is not passed" comment is ambiguous. Is that the end goal or just a description of the current, intermediate state? A better comment is "Logger is used for background activities".

This is description of the current state. Storing of the logger is because the whole code base is not yet transitioned. This is therefore temporary and will be removed once the context is being passed down. 
I have updated the comment as per your suggestion.

Reply:
I'd propose to look at all interface methods and try to pass context there. If it can't be passed then using a stored logger is acceptable.

For example here is what I see for the GetMemory method:
context can be passed this way: `v1PodResourcesServer.List` or `v1PodResourcesServer.Get` -> `v1PodResourcesServer.getContainerResources `-> container manager `GetMemory` -> memory manager `GetMemory`

If this kind of changes touch a lot of components, it's ok to use context.TODO at some point(in the example above I'd use it in the container manager or even in the memory manager `GetMemory` method) with a description how a context is going to be passed in future.

Reply:
@bart0sh Please review https://github.com/kubernetes/kubernetes/pull/130727/commits/360ffdfc96db737b92db1e12c3fd525715582699 commit and let me know if it addresses your review comment. I was able to plumb in context into `GetAllocatableMemory` and `GetMemory` with context.TODO() passed from container manager temporarily. 
This could not be done for all the methods as that would require changes to other resource managers and is outside the scope of this PR.

If you are happy enough, I will go ahead and squash the commits to cleanup the git history.

Reply:
Done, Thanks!